packages/theme-check-common/src/checks/unique-settings-id/index.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  isArrayNode,
+  isLiteralNode,
+  isObjectNode,
+  isPropertyNode,
+  Severity,
+  SourceCodeType,
+} from '../../types';
+
+import type { ArrayNode, PropertyNode, JSONCheckDefinition } from '../../types';
+
+export const UniqueSettingIds: JSONCheckDefinition = {
+  meta: {
+    code: 'UniqueSettingId',
+    name: 'Prevent duplicate Ids in setting_schema',
+    docs: {
+      description: 'This check is aimed at eliminating duplicate Ids in settings_schema.json',
+      recommended: true,
+      // url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/valid-schema',
+    },
+    type: SourceCodeType.JSON,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    return {
+      async onCodePathEnd(file) {
+        if (isArrayNode(file.ast)) {
+          const settingIds: PropertyNode[] = [];
+
+          /* Find and loop through all of our nodes that have an id value and find their key value */
+          for (const child of file.ast.children) {
+            if (isObjectNode(child) && child.children) {
+              const settingsNode = child.children.find((node) => node.key.value === 'settings');
+
+              if (settingsNode && settingsNode.value && isArrayNode(settingsNode.value)) {
+                for (const setting of settingsNode.value.children) {
+                  if (isObjectNode(setting) && setting.children) {
+                    const idNode = setting.children.find((node) => node.key.value === 'id');
+                    if (isPropertyNode(idNode)) {
+                      settingIds.push(idNode);
+                    }
```

Comment:
We could also move most of the "Check for dupes" code here instead of using a "settingIds" variable.

I guess you didn't want to make the code here too nested and dirty 😅

```suggestion
                    if (isPropertyNode(idNode) && isLiteralNode(idNode.value)) {
                      const id = idNode.value.value;
                      if (typeof id === 'string') {
                        if (!idMap.has(id)) {
                          idMap.set(id, []);
                        }
                        idMap.get(id)!.push(idNode);
                      }
                    }
```

Reply:
@aswamy yep spot on, not a fan of nests and rather use guard clauses but due to the nature of this particular paradigm guard clauses don't fit. Didn't see a style guide from Shopify anywhere either.